UEBPrep/app/controllers/cards_controller.rb

```diff
@@ -7,11 +7,23 @@ class CardsController < ApplicationController
   # GET /cards
   def index
     @cards = Card.order(:id).page params[:page]
+    respond_to do |format|
+      format.html
+      format.json { render json: @cards }
+    end
   end
 
   # GET /cards/1
   def show
-    @current_user = current_user
+    unless session.blank?
+      @current_user = current_user
```

Comment:
Because we have public cards, we don't always need a current user for this action.